distributed/tests/test_client_executor.py

```diff
@@ -218,30 +227,32 @@ def test_retries(client):
         exc_info.match("one")
 
 
-def test_shutdown(client):
-    # shutdown(wait=True) waits for pending tasks to finish
-    e = client.get_executor()
-    fut = e.submit(time.sleep, 1.0)
-    t1 = time.time()
-    e.shutdown()
-    dt = time.time() - t1
-    assert 0.5 <= dt <= 2.0
-    time.sleep(0.1)  # wait for future outcome to propagate
-    assert fut.done()
-    fut.result()  # doesn't raise
-
-    with pytest.raises(RuntimeError):
-        e.submit(time.sleep, 1.0)
-
-    # shutdown(wait=False) cancels pending tasks
-    e = client.get_executor()
-    fut = e.submit(time.sleep, 2.0)
-    t1 = time.time()
-    e.shutdown(wait=False)
-    dt = time.time() - t1
-    assert dt < 0.5
-    time.sleep(0.1)  # wait for future outcome to propagate
-    assert fut.cancelled()
-
-    with pytest.raises(RuntimeError):
-        e.submit(time.sleep, 1.0)
+def test_shutdown(loop):
+    with cluster(disconnect_timeout=10) as (s, [a, b]):
```

Comment:
I'm happy to see this test just because I know that it means that you dove into the utils_test.py infrastructure.  Hooray for increasing bus factors!

Reply:
`bus_factor += 1`